math/iterative_factorial.cpp

```diff
@@ -0,0 +1,116 @@
+/**
+ * @file
+ * @brief Iterative implementation of factorial
+ *
+ * @author [Renjian-buchai](https://github.com/Renjian-buchai)
+ *
+ * @description Calculates factorial iteratively.
+ * https://en.wikipedia.org/wiki/Factorial
+ *
+ * Factorial formula:
+ * n! = n * (n - 1) * (n - 2) * ... * 2 * 1
+ *
+ * @example
+ *
+ * 5! = 5 * 4 * 3 * 2 * 1
+ *
+ * Recursive implementation of factorial pseudocode:
+ *
+ * function factorial(n):
+ *     if n == 1:
+ *         return 1
+ *     else:
+ *         return factorial(n-1)
+ *
+ */
+
+#include <cassert>    // For assert
+#include <exception>  // For std::invalid_argument
+#include <iostream>   // for std::cout
```

Comment:
```suggestion
#include <cassert>    /// for assert
#include <exception>  /// for std::invalid_argument
#include <iostream>   /// for std::cout
```
triple slashes for includes!